### PR TITLE
changes for issues/85

### PR DESCRIPTION
**Pull request by tdiepenbrock**

### CODE REVIEW

File: src/test/ml-data/feature-services/test/DataSourceArray.json [hunk 1]
```diff
@@ -469,6 +469,246 @@
...
+          "latestWkid": 4326
+        }
+      },
+      "dataSources": [
+        {
+          "source": "view",
+          "schema": "GDeltGKG",
+          "view": "Article",
+          "fragmentIdColumn": "FRAGMENT_ID_Article",
+          "joinFunction": "joinLeftOuter",
```
Comment (jkerr5) [inline src/test/ml-data/feature-services/test/DataSourceArray.json:496]: Do we need the `joinFunction` on this data source? It seems like we just want to have the `joinOn` object in the next data source.

File: src/test/ml-data/feature-services/test/DataSourceArray.json [hunk 2]
```diff
@@ -469,6 +469,246 @@
...
+          "includeFields": [
+            "OBJECTID"
+          ]
+        },
+        {
+          "source": "view",
+          "schema": "GdeltGKG",
+          "view": "Article_Join",
+          "fragmentIdColumn": "FRAGMENT_ID_Article_Join",
+          "joinFunction": "joinLeftOuter",
```
Comment (jkerr5) [inline src/test/ml-data/feature-services/test/DataSourceArray.json:511]: Can this be put inside the `joinOn` object? Maybe 
```
joinOn {
  left
  right
  joinFunction
}
```
Maybe `joinFunction` could be just `function` if inside the `joinOn` object?

**Review by jkerr5 — COMMENTED** [review]
A few thoughts on where the `joinFunction` should live in the descriptor but otherwise, looks good.
Of course, some docs in the wiki would also be good but at least there are examples in the test cases.

**Comment by tdiepenbrock** [comment]
After further discussions:

We will move the type of join into the joinOn property.  The type of join will be indicated by a “joinType” property, which can have legal values of “inner”, “left outer”, and “full outer”:


joinOn {
  “left”:”leftColumn”,
  “right”:”rightColumn”,
  “joinType”:[“inner” | “left outer” | “full outer"]
}


Thomas Diepenbrock
Tech Director, Federal Consulting
MarkLogic Corporation<http://www.marklogic.com>
thomas.diepenbrock@marklogic.com<mailto:thomas.diepenbrock@marklogic.com>
Office: 1+703.854.8510
Cell: 1+703.980.1248

[cid:11161830-1938-4A05-B9D4-9E69F42612B5]

On Feb 3, 2021, at 10:22 AM, James Kerr <notifications@github.com<mailto:notifications@github.com>> wrote:



@jkerr5 commented on this pull request.

A few thoughts on where the joinFunction should live in the descriptor but otherwise, looks good.
Of course, some docs in the wiki would also be good but at least there are examples in the test cases.

________________________________

In src/test/ml-data/feature-services/test/DataSourceArray.json<https://github.com/marklogic-community/marklogic-geo-data-services/pull/86#discussion_r569497097>:

> +        "xmin": -180,
+        "ymin": -90,
+        "xmax": 180,
+        "ymax": 90,
+        "spatialReference": {
+          "wkid": 4326,
+          "latestWkid": 4326
+        }
+      },
+      "dataSources": [
+        {
+          "source": "view",
+          "schema": "GDeltGKG",
+          "view": "Article",
+          "fragmentIdColumn": "FRAGMENT_ID_Article",
+          "joinFunction": "joinLeftOuter",


Do we need the joinFunction on this data source? It seems like we just want to have the joinOn object in the next data source.

________________________________

In src/test/ml-data/feature-services/test/DataSourceArray.json<https://github.com/marklogic-community/marklogic-geo-data-services/pull/86#discussion_r569499707>:

> +          "joinFunction": "joinLeftOuter",
+          "fields": {
+            "OBJECTID": {
+            "scalarType": "long"
+            }
+          },
+          "includeFields": [
+            "OBJECTID"
+          ]
+        },
+        {
+          "source": "view",
+          "schema": "GdeltGKG",
+          "view": "Article_Join",
+          "fragmentIdColumn": "FRAGMENT_ID_Article_Join",
+          "joinFunction": "joinLeftOuter",


Can this be put inside the joinOn object? Maybe

joinOn {
  left
  right
  joinFunction
}


Maybe joinFunction could be just function if inside the joinOn object?

**Comment by tdiepenbrock** [comment]
After further discussions:

We will move the type of join into the joinOn property.  The type of join will be indicated by a “joinType” property, which can have legal values of “inner”, “left outer”, and “full outer”:

joinOn {
  “left”:”leftColumn”,
  “right”:”rightColumn”,
  “joinType”:[“inner” | “left outer” | “full outer"]
}

**Review by jkerr5 — APPROVED** [review]
Look good to me

**Comment by jkerr5** [comment]
Closes issue #85

**Pull request merged**